src/System.Text.Json/tests/Serialization/Array.ReadTests.cs

```diff
@@ -427,14 +427,16 @@ public static void ClassWithNoSetter()
 
         public class ClassWithPopulatedListAndSetter
         {
-            public List<int> MyList { get; set;  } = new List<int>() { 1 };
+            public List<int> MySkippedList { get; }
```

Comment:
Can you add a test where the skipped list property is ordered between two non-skipped properties?

Also, add one where `ClassWithPopulatedListAndSetter` is nested within another POCO class which has skipped/non-skipped properties, primarily to increase confidence that the fix works for edge cases as well.

Reply:
@ahsonkhan I added a test that mixes everything together. I made it as complicated as my creativity would allow!